File: Demo/NUITests/UIKitTests/UITableViewCell/NUITableViewCellRendererTests.m
//
//  NUITableViewCellRendererTests.m
//  NUIDemo
//
//  Created by Alexey Trenikhin on 3/14/16.
//  Copyright © 2016 Tom Benner. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "NUIConstants.h"
#import "NUIFileMonitor.h"
#import "NUISettings.h"
#import "UIView+NUI.h"

#import "NUIDemo-Swift.h"

@interface NUITableViewCellRendererTests : XCTestCase
@property (strong, nonatomic) UITableViewCell *sut;
@end

@implementation NUITableViewCellRendererTests

- (void)setUp {
    [super setUp];
    [NUISettings initWithStylesheet:@"TestTheme.NUI"];
    _sut = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"test"];
    [_sut applyNUI];
}

- (void)tearDown {
    self.sut = nil;
    [super tearDown];
}

- (void)testContentViewBackgroundColor {
    XCTAssertEqualObjects(self.sut.contentView.backgroundColor, [UIColor orangeColor], @"NUI should set content view background color");
}

- (void)testTintColor {
    XCTAssertEqual(_sut.tintColor, [UIColor greenColor] , @"NUI should set tintColor");
}


@end

File: Demo/NUITests/UIKitTests/UITableViewCell/NUITableViewCellRendererTests.swift
//
//  NUITableViewCellRendererTests.swift
//  NUIDemo
//
//  Created by Sergej Jaskiewicz on 04/08/16.
//  Copyright © 2016 Tom Benner. All rights reserved.
//

import XCTest
@testable import NUIDemo

class NUITableViewCellRendererTests: XCTestCase {
    
    var sut: UITableViewCell!
    
    override func setUp() {
        super.setUp()
        
        NUISettings.initWithStylesheet("TestTheme.NUI")
        sut = UITableViewCell(style: .default, reuseIdentifier: "test")
        sut.applyNUI()
    }
    
    func testContentViewBackgroundColor() {
        XCTAssertEqual(sut.contentView.backgroundColor, .orange, "NUI should set content view background color")
    }
    
    func testTintColor() {
        XCTAssertEqual(sut.tintColor, .green, "NUI should set tintColor")
    }
}
